workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/AdministratorDashboard/hooks/useAccountsMetrics.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {addDays} from 'date-fns';
+import useSWR from 'swr';
+
+import SearchBuilder from '../../../core/SearchBuilder';
+import HeadlessAdminUserImpl from '../../../services/rest/HeadlessAdminUser';
+
+export const METRIC_PARAMETER = {
+	month: 30,
+	q1: 1,
+	q2: 2,
+	q3: 3,
+	q4: 4,
+	week: 7,
+};
+
+type useOrderMetricsProps = 'month' | 'q1' | 'q2' | 'q3' | 'q4' | 'week';
+
+const useAccountsMetrics = (param: useOrderMetricsProps) => {
+	const getAccountsMetrics = async () => {
+		const currentTime = new Date();
+
+		const lastPeriod = addDays(
+			currentTime,
+			-METRIC_PARAMETER[param as keyof typeof METRIC_PARAMETER]
+		).toISOString();
+
+		const beforeLastPeriod = addDays(
+			currentTime,
+			-METRIC_PARAMETER[param as keyof typeof METRIC_PARAMETER] * 2
+		).toISOString();
+
+		const requestsParams = [
+			{
+				searchParams: new URLSearchParams({
+					fields: 'id,',
+					pageSize: '-1',
+				}),
+			},
+			{
+				searchParams: new URLSearchParams({
+					fields: 'id,',
+					filter: SearchBuilder.gt('dateCreated', lastPeriod),
+				}),
+			},
+			{
+				searchParams: new URLSearchParams({
+					fields: 'id,',
+					filter: new SearchBuilder()
+						.gt('dateCreated', lastPeriod)
+						.and()
+						.lt('dateCreated', beforeLastPeriod)
+						.build(),
+				}),
+			},
```

Comment:
No needs for this to be an object

```suggestion
			{
				searchParams: new URLSearchParams({
					fields: 'id,',
					pageSize: '-1',
				}),
			},
			{
				searchParams: new URLSearchParams({
					fields: 'id,',
					filter: SearchBuilder.gt('dateCreated', lastPeriod),
				}),
			},
			{
				searchParams: new URLSearchParams({
					fields: 'id,',
					filter: new SearchBuilder()
						.gt('dateCreated', lastPeriod)
						.and()
						.lt('dateCreated', beforeLastPeriod)
						.build(),
				}),
			},
```